pkg/resources/saml_integration.go

```diff
@@ -0,0 +1,476 @@
+			log.Printf("[WARN] unexpected security integration property %v returned from Snowflake", k)
+		}
+	}
+
+	return err
+}
+
+// UpdateSAMLIntegration implements schema.UpdateFunc
+func UpdateSAMLIntegration(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	id := d.Id()
+
+	stmt := snowflake.SamlIntegration(id).Alter()
+
+	var runSetStatement bool
+
+	if d.HasChange("enabled") {
+		runSetStatement = true
+		stmt.SetBool(`ENABLED`, d.Get("enabled").(bool))
+	}
+
+	if d.HasChange("saml2_issuer") {
+		runSetStatement = true
+		stmt.SetString(`SAML2_ISSUER`, d.Get("saml2_issuer").(string))
+	}
+
+	if d.HasChange("saml2_sso_url") {
+		runSetStatement = true
+		stmt.SetString(`saml2_sso_url`, d.Get("saml2_sso_url").(string))
+	}
+
+	if d.HasChange("saml2_provider") {
+		runSetStatement = true
+		stmt.SetString(`saml2_provider`, d.Get("saml2_provider").(string))
+	}
+
+	if d.HasChange("saml2_x509_cert") {
+		runSetStatement = true
+		stmt.SetString(`saml2_x509_cert`, d.Get("saml2_x509_cert").(string))
+	}
+
+	if d.HasChange("saml2_sp_initiated_login_page_label") {
+		runSetStatement = true
+		stmt.SetString(`saml2_sp_initiated_login_page_label`, d.Get("saml2_sp_initiated_login_page_label").(string))
+	}
+
+	if d.HasChange("saml2_enable_sp_initiated") {
+		runSetStatement = true
+		stmt.SetBool(`saml2_enable_sp_initiated`, d.Get("saml2_enable_sp_initiated").(bool))
+	}
+
+	if d.HasChange("saml2_snowflake_x509_cert") {
+		runSetStatement = true
+		stmt.SetString(`saml2_snowflake_x509_cert`, d.Get("saml2_snowflake_x509_cert").(string))
+	}
+
+	if d.HasChange("saml2_sign_request") {
+		runSetStatement = true
+		stmt.SetBool(`saml2_sign_request`, d.Get("saml2_sign_request").(bool))
+	}
+
+	if d.HasChange("saml2_requested_nameid_format") {
+		runSetStatement = true
+		stmt.SetString(`saml2_requested_nameid_format`, d.Get("saml2_requested_nameid_format").(string))
+	}
+
+	if d.HasChange("saml2_post_logout_redirect_url") {
+		runSetStatement = true
+		stmt.SetString(`saml2_post_logout_redirect_url`, d.Get("saml2_post_logout_redirect_url").(string))
+	}
+
+	if d.HasChange("saml2_force_authn") {
+		runSetStatement = true
+		stmt.SetBool(`saml2_force_authn`, d.Get("saml2_force_authn").(bool))
+	}
+
+	if d.HasChange("saml2_snowflake_issuer_url") {
+		runSetStatement = true
+		stmt.SetString(`saml2_snowflake_issuer_url`, d.Get("saml2_snowflake_issuer_url").(string))
+	}
+
+	if d.HasChange("saml2_snowflake_acs_url") {
+		runSetStatement = true
+		stmt.SetString(`saml2_snowflake_acs_url`, d.Get("saml2_snowflake_acs_url").(string))
+	}
+
+	if runSetStatement {
+		if err := snowflake.Exec(db, stmt.Statement()); err != nil {
+			return errors.Wrap(err, "error updating security integration")
+		}
+	}
+
+	return ReadSAMLIntegration(d, meta)
+}
+
+// DeleteSAMLIntegration implements schema.DeleteFunc
+func DeleteSAMLIntegration(d *schema.ResourceData, meta interface{}) error {
+	return DeleteResource("", snowflake.ScimIntegration)(d, meta)
```

Comment:
`snowflake.ScimIntegration` seems off -- is this snowflake.SAMLIntegration?

Reply:
good catch, fixed in latest commit

Reply:
thank you!